lib/matplotlib/cbook.py

```diff
@@ -2445,6 +2445,14 @@ def safe_first_element(obj):
     return next(iter(obj))
 
 
+def asSequence(data):
```

Comment:
Hmm...even with duct typing, there's some basic something they need to have though that generators and views don't...granted, I'm also fine with the less work don't change anything and just find a good name approach...